frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,561 @@
+      encryptionCount: BigNumber.from(2).pow(255).toString(),
+      expectedDecryptedData: {
+        advancedMode: false,
+        usePublicKeyChecked: true,
+        address: recipientAddress,
+        pubKey: partialPubKey,
+      },
+    },
+  ].forEach(({ ciphertext, encryptionCount, expectedDecryptedData }) => {
+    it('Data decrypted correctly all true', () => {
+      const data = decryptData(ciphertext, { encryptionCount, viewingPrivateKey: viewingPrivateKey });
+      expect(data).toEqual(expectedDecryptedData); // Recursively checks all properties.
+    });
+  });
+});
+
+describe('encrypt/decrypt relationship', () => {
+  it('Encrypt and decrypt are inverses', () => {
+    // Generate random data and do multiple runs.
+    for (let i = 0; i < 100; i++) {
+      const { recipientAddress, pubKey, viewingPrivateKey, advancedMode, usePublicKeyChecked, encryptionCount } =
+        randomData();
+      const ciphertext = encryptAccountData(
+        { recipientAddress, advancedMode, pubKey, usePublicKeyChecked },
+        { encryptionCount, viewingPrivateKey }
+      );
+      const decryptedData = decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+      expect(decryptedData).toEqual({
+        advancedMode,
+        usePublicKeyChecked,
+        address: recipientAddress,
+        pubKey: `0x99${pubKey.slice(4, 4 + 22)}`,
+      });
+    }
+  });
+
+  it('Encrypts and decrypts properly for values with leading zeros', () => {
+    // `BigNumber.from(myHexString).toHexString()` where `myHexString` has leading zero bytes will
+    // truncate those leading zeroes from the string. If data is not manually zero-padded to the
+    // correct length, this can cause a bug where the leading zero bytes of the address are
+    // truncated and trailing bytes of other data are appended. The above fuzz test found this bug,
+    // and this is a concrete test with known inputs would fail if the bug were present.
+    const recipientAddress = '0x00bB833415cf56f112389E146Cb7847dDbE93fB5';
+    const pubKey =
+      '0x040426c84897f9e07632687f69ef714090307dcdc23b4a9f3719e6db8eb1f753f71dd4f849c6101d13049f6091944914c31e6fe32db34daa176da230a04de12bf3';
+    const viewingPrivateKey = '0xeaa492b979aead8bdaf857673c4fe453a3f7ce5e8d661499391505f47d96d613';
+    const advancedMode = false;
+    const usePublicKeyChecked = false;
+    const encryptionCount = '661';
+
+    const ciphertext = encryptAccountData(
+      { recipientAddress, advancedMode, pubKey, usePublicKeyChecked },
+      { encryptionCount, viewingPrivateKey }
+    );
+    const decryptedData = decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+    expect(decryptedData).toEqual({
+      advancedMode,
+      usePublicKeyChecked,
+      address: recipientAddress,
+      pubKey: `0x99${pubKey.slice(4, 4 + 22)}`,
+    });
+  });
+});
+
+describe('storeSend', () => {
+  const testingUtils = generateTestingUtils({ providerType: 'default' });
```

Comment:
Do we need this new dependency, or can we remove it and instantiate a real ethers provider like we do in the umbra-js tests, since we have RPC URL env vars for the frontend already?

Reply:
Do we **need** the dependency, definitely not. We could use the RPC url. The benefits I see from using this mock are that the tests remain unit tests and we aren't using rpc calls when running the tests locally. I think the tradeoff is worth it, but am not opposed to switching it.

Reply:
Yes, exactly to fetch addresses from ENS. We could split and handle in the account sent table if that is better.

Reply:
Feels like a code smell that we can't have an e2e fuzz test without a provider. I guess the mock provider is fine though, I just like to not introduce deps when avoidable to mitigate the risk of nodejs supply chain attacks

Reply:
Yes, I have that concern as well. It probably makes the most sense to split out the ens logic and handle it in the component that renders the data. The code is unnecessarily coupled.

Reply:
Actually, a simpler solution was to use the MAINNET_PROVIDER instead.

https://github.com/ScopeLift/umbra-protocol/pull/494/commits/14814552f74b1f38c932b1e765a2d02eeb205d46

Reply:
nice good point, I like that